prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -386,7 +386,13 @@ open class BaseHistoryFunction : Logging {
                         }
                     )
                     .content(String(contents))
-                    .fileName(requestedReport.externalName)
+                    .fileName(
+                        if (requestedReport.externalName.isNullOrBlank()) {
```

Comment:
This will "fix" the bug, unclear if it really matters that much what the file name is?